staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -344,16 +348,39 @@ func (o LogsOptions) parallelConsumeRequest(requests []rest.ResponseWrapper) err
 	return err
 }
 
-func (o LogsOptions) sequentialConsumeRequest(requests []rest.ResponseWrapper) error {
-	for _, request := range requests {
-		if err := o.ConsumeRequestFn(request, o.Out); err != nil {
+func (o LogsOptions) sequentialConsumeRequest(requests map[corev1.ObjectReference]rest.ResponseWrapper) error {
+	for objRef, request := range requests {
+		out := o.addPrefixIfNeeded(objRef, o.Out)
+		if err := o.ConsumeRequestFn(request, out); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func (o LogsOptions) addPrefixIfNeeded(ref corev1.ObjectReference, writer io.Writer) io.Writer {
+	if !o.Prefix {
+		return writer
+	}
+
+	// We rely on ref.FieldPath to contain a reference to a container
+	// including a container name (not an index) so we can get a container name
+	// without making an extra API request.
+	var containerName string
+	re := regexp.MustCompile(`spec\.(?:initContainers|containers|ephemeralContainers){(.+)}`)
+	containerNameMatches := re.FindStringSubmatch(ref.FieldPath)
+	if len(containerNameMatches) == 2 {
+		containerName = containerNameMatches[1]
+	}
+
+	prefix := fmt.Sprintf("[pod/%s -c %s] ", ref.Name, containerName)
```

Comment:
`pod/<pod-name>/<container-name>` would be the best imo.

Reply:
@soltysh  I can do that. The reason why I decided to go with `pod/%s -c %s` is becasue it's easy to copy-paste this bit into a new `kubectl logs` call (in a separate window). It was very handy while working on the PR itself.

But I do not have strong preferences here.